solution/index.js

```diff
@@ -0,0 +1,585 @@
+        ul = e.target.parentElement.previousElementSibling;
+    }
+    const listType = taskInput.closest("section").dataset.listType;
+
+    if(!taskInput.value) {
+        alert("Invalid input!");
+        return;
+    }
+
+    let reminder = null;
+    if(reminderInput.value) {
+        reminder = new Date(reminderInput.value).toLocaleString();
+    }
+
+    const task = {
+        task: taskInput.value,
+        date: new Date().toLocaleString(),
+        reminder: reminder,
+    }
+
+    const reminderInMilliseconds = (new Date(task.reminder).getTime() - new Date(task.date).getTime());
+    if(reminder && reminderInMilliseconds > 0) {
+        setTimeout(() => {
+            alert(`Reminder from "${listType}" list:\n"${task.task}"`);
+        }, reminderInMilliseconds);
+    }
+
+    addTask(task, listType);
+    
+    taskInput.value = "";
+    reminderInput.value = "";
+    reminderInput.type = "text";
+
+    generateLists();
+}
+
+//Handles the task edit
+function editTaskEventHandler(e) {
+    if(e.target.tagName !== "LI") return;
+
+    const liTask = e.target;
+    liTask.contentEditable = true;
+    liTask.draggable = false;
+    liTask.classList.add("text-cursor");
+    liTask.focus();
+
+    const listType = liTask.closest("section").dataset.listType;
+
+    const newTask = {
+        task: liTask.textContent,
+        date: new Date().toLocaleString(),
+        reminder: liTask.dataset.reminder,
+    }
+
+    const oldTask = {
+        task: liTask.dataset.task,
+        date: liTask.dataset.date,
+        reminder: liTask.dataset.reminder,
+    }
+
+    if(e.key === "Enter") {
+        updateTask(newTask, oldTask, listType);
+        generateLists()
+    }
+}
+
+//Handles the blur event so if the task element loses focus, the change will be saved.
+function blurEventHandler(e) {
+    if(e.target.tagName === "LI") {
+        const liTask = e.target;
+        liTask.contentEditable = false;
+        liTask.draggable = true;
+        liTask.classList.remove("text-cursor");
+        
+        const listType = liTask.closest("section").dataset.listType;
+
+        const newTask = {
+            task: liTask.textContent,
+            date: new Date().toLocaleString(),
+            reminder: liTask.dataset.reminder,
+        }
+    
+        const oldTask = {
+            task: liTask.dataset.task,
+            date: liTask.dataset.date,
+            reminder: liTask.dataset.reminder,
+        }
+        
+        updateTask(newTask, oldTask, listType);
+
+        generateLists();
+    }
+}
+
+//Handles the task movements between lists
+function keydownMoveTaskEventHandler(e) {
+    const mouseOverElements = document.querySelectorAll(":hover");
+    const liMouseOn = mouseOverElements[mouseOverElements.length - 1];
+
+    if(!liMouseOn || liMouseOn.tagName !== "LI" || e.key !== 'Alt') return;
+
+    altKeyDownEventHandler(e);
+}
+function altKeyDownEventHandler(e) {
```

Comment:
There is an easy way to check whether the alt key was pressed during a keyboard event. You don't need extra event handlers.